### PR TITLE
Qualify did exch connection lookup by role

### PR DESCRIPTION
**Pull request by ianco**
Signed-off-by: Ian Costanzo <ian@anon-solutions.ca>

Fixes issue #1541

### CODE REVIEW

File: aries_cloudagent/protocols/didexchange/v1_0/manager.py [hunk 1]
```diff
@@ -702,13 +702,22 @@ async def accept_response(
...
                     )
-            except StorageNotFoundError:
-                pass
+                except StorageNotFoundError:
+                    pass
+                if not conn_rec:
+                    try:
+                        conn_rec = await ConnRecord.retrieve_by_request_id(
+                            session, response._thread_id,
+                            their_role=ConnRecord.Role.RESPONDER.rfc160,
```
Comment (ianco) [inline aries_cloudagent/protocols/didexchange/v1_0/manager.py:717]: It looks like the rfc23 roles should be used, but the created connection records are using the rfc160 roles ... so I included checks for both

**Review by TimoGlastra — APPROVED** [review]
Great! Thanks @ianco

Adding `their_role` to the `TAG_NAMES` means this will prevent old exchanges from finishing after PR has been released. This is not an issue for completed or new exchanges so I guest it's not really an issue? 

Good to keep track somehow of later added tags, as now it may look you can query record by the `their_role` property, but this is only the case for records saved _after_ this PR has been merged.

**Comment by swcurran** [comment]
We do have the "migrate" mechanism to update stored data on deployment of a new version, but given the tight scope of this change, I don't think we need to use it.  Agreed?

We used the mechanism with 0.73 to update connection records.

**Comment by ianco** [comment]
I don't think we need to "migrate" for this function because it's only used during the process of establishing a connection.

Once the connection is established we don't use the "request_id" to lookup the connection.

**Pull request merged**